fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/DataServlet.java

```diff
@@ -451,17 +452,18 @@ public void getData(
             if (!complete) {
                 throw new IllegalStateException("No request handler accepted request.");
             }
-        } catch (RequestValidationException e) {
-            LOG.debug(e.getMessage(), e);
-            asyncResponse.resume(RequestHandlerUtils.makeErrorResponse(e.getStatus(), e, writer));
-        } catch (NoMatchFoundException e) {
-            LOG.info("Exception processing request", e);
-            asyncResponse.resume(RequestHandlerUtils.makeErrorResponse(INTERNAL_SERVER_ERROR, e, writer));
-        } catch (TimeoutException e) {
-            LOG.info("Exception processing request", e);
-            asyncResponse.resume(RequestHandlerUtils.makeErrorResponse(GATEWAY_TIMEOUT, e, writer));
-        } catch (Error | Exception e) {
-            LOG.info("Exception processing request", e);
+        } catch (Throwable e) {
+            exceptionHandler.handleException(
+                e,
+                asyncResponse,
+                Optional.ofNullable(apiRequest),
+                containerRequestContext,
+                uriInfo,
+                writer
+            );
+            // Generally, it's expected that implementations of `ExceptionHandler` will resume
+            // the response in every case. This exists so that if someone writes a buggy handler
+            // that fails to resume the response, they at least get *something* back.
```

Comment:
The 'just in case' asynch response resume should check if the response is already done and if it /isn't/ send a message indicating that an error was mishandled.  (A 500 error, probably, also passing along the error that was badly handled)